launch/launch/utilities/ensure_argument_type_impl.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module for the ensure_argument_type() utility function."""
+
+import collections
+import inspect
+from typing import Any
+from typing import Iterable
+from typing import Optional
+from typing import Text
+from typing import Union
+
+
+def ensure_argument_type(
+    argument: Any,
+    types: Union[type, Iterable[type]],
+    argument_name: Text,
+    caller: Optional[Text] = None,
+) -> None:
+    """
+    Ensure that the given argument is an instance of or subclass of one of the given types.
+
+    If the argument does not match one of the types, a TypeError is raised.
+    The caller is included in the error message if given and not None.
+    """
+    error_msg_template = "{}xpected '{}' to be one of [{}], but got '{}' of type '{}'"
+    if not isinstance(types, collections.Iterable) and not isinstance(types, type):
```

Comment:
I think it's right, though it could be rewritten as `ininstance(types, type(type))`. Basically `types` either needs to be an iterable (of types) or a single type. I think it's the same signature as `isinstance()`, e.g. both `isinstance(1, int)` and `isinstance(1, (int, float))` are valid.

Reply:
I didn't realise `type` could be this way because I usually use it as if it were function. Pardon the noise! We might want to check though that in the iterable case it's actually an iterable of types, but that would just be to get a nicer error message.

Reply:
It's fine, not like it's completely obvious :)

If you're ok with it, I'll make an issue to improve this check, because as you said, it's only a nicer error message.